packages/toolpad-app/src/toolpad/AppEditor/NodeMenu.tsx

```diff
@@ -88,12 +87,14 @@ export default function NodeMenu({
           menuProps.onClick?.(event);
         }}
       >
-        <MenuItem onClick={handleRenameClick}>
-          <ListItemIcon>
-            <ModeEditIcon />
-          </ListItemIcon>
-          <ListItemText>{renameLabelText}</ListItemText>
-        </MenuItem>
+        {onRenameNode ? (
```

Comment:
Should duplicate and delete also have this condition?

Reply:
Yep, makes sense